lib/Doctrine/ORM/Proxy/ProxyFactory.php

```diff
@@ -97,23 +160,40 @@ public function __serialize(): array
      */
     public function __construct(EntityManagerInterface $em, $proxyDir, $proxyNs, $autoGenerate = self::AUTOGENERATE_NEVER)
     {
-        $proxyGenerator = new ProxyGenerator($proxyDir, $proxyNs);
-
-        if ($em->getConfiguration()->isLazyGhostObjectEnabled()) {
-            $proxyGenerator->setPlaceholder('baseProxyInterface', InternalProxy::class);
-            $proxyGenerator->setPlaceholder('useLazyGhostTrait', Closure::fromCallable([$this, 'generateUseLazyGhostTrait']));
-            $proxyGenerator->setPlaceholder('skippedProperties', Closure::fromCallable([$this, 'generateSkippedProperties']));
-            $proxyGenerator->setPlaceholder('serializeImpl', Closure::fromCallable([$this, 'generateSerializeImpl']));
-            $proxyGenerator->setProxyClassTemplate(self::PROXY_CLASS_TEMPLATE);
-        } else {
+        if (! $em->getConfiguration()->isLazyGhostObjectEnabled()) {
+            if (PHP_VERSION_ID >= 80100) {
+                Deprecation::trigger(
+                    'doctrine/orm',
+                    'https://github.com/doctrine/orm/pull/10837/',
+                    'Not enabling lazy ghost objects is deprecated and will not be supported in Doctrine ORM 3.0. Ensure Doctrine\ORM\Configuration::setLazyGhostObjectEnabled(true) is called to enable them.'
```

Comment:
We should deprecate calling `Configuration::setLazyGhostObjectEnabled(false)` as well because `true` is going to be the only valid setting in 3.0.

Reply:
It's kinda already the case, because if you set this to false, you'll get the deprecation added to ProxyFactory.

Reply:
I see.